components/tools/OmeroWeb/omeroweb/webgateway/views.py

```diff
@@ -1198,6 +1210,40 @@ def wrap(request, *args, **kwargs):
     return wrap
 
 
+def json_response(f):
+    """
+    Decorator for wrapping response in JsonResponse if needed and
+    error handling
+
+    @param f:       The function to wrap
+    @return:        The wrapped function, which will return json
+    """
+    @wraps(f)
+    def wrap(request, *args, **kwargs):
+        logger.debug('json_response')
+        try:
+            rv = f(request, *args, **kwargs)
+            if isinstance(rv, HttpResponse):
```

Comment:
I don't see how my comment prevent from handling exception in a view. This is what Exceptions are for. It is just fundamental mistake. A function has a "type" -- inputs mapped to outputs. It **must return** a consistent type of result, or it isn't a function. If can't return a proper result, raise an exception.